src/LayerEditor/ui/layers_panel/dialogs/split_layer_dlg.py

```diff
@@ -1,21 +1,24 @@
 """
 Dialog for appending new layer to the end.
 """
+import math
+import numpy as np
 
 import PyQt5.QtWidgets as QtWidgets
 import PyQt5.QtGui as QtGui
 from PyQt5.QtCore import Qt, QPoint
 
 import gm_base.icon as icon
+from LayerEditor.ui.layers_panel import helper_functions
 
 
 class SplitLayerDlg(QtWidgets.QDialog):
 
-    def __init__(self, top_y, bot_y, used_names, parent=None):
+    def __init__(self, top_y, bot_y, le_model, parent=None):
         super(SplitLayerDlg, self).__init__(parent)
-        self.used_names = used_names
-        self.top_y = top_y if top_y is not None else 10000
-        self.bot_y = bot_y if bot_y is not None else -10000
+        self.le_model = le_model
```

Comment:
It would be enough if would make the two methods again. DRY. If the le_model is so bad here than I could pass the two methods, but it would not help readability.

Reply:
is_unique_layer_name vs. is_layer_name_unique